rotkehlchen/premium/premium.py

```diff
@@ -273,12 +287,11 @@ def upload_data(
             log.error(msg)
             raise RemoteError(msg) from e
 
-        check_response_status_code(
+        return _process_dict_response(
```

Comment:
This is why I made `check_response_status_code` and `process_dict_response` have same arguments. Before we were calling them both here and as a result since `check_response_status_code` also happens in `process_dict_response` the check happened twice.